standardtracer/span.go

```diff
@@ -11,22 +11,33 @@ import (
 // Implements the `Span` interface. Created via tracerImpl (see
 // `standardtracer.New()`).
 type spanImpl struct {
-	lock     sync.Mutex
-	tracer   *tracerImpl
-	recorder SpanRecorder
-	raw      RawSpan
+	tracer     *tracerImpl
+	sync.Mutex // protects the fields below
```

Comment:
I'm fine with the decision, I was just curious if it came from a specific benchmark or more of an ad hoc hypothesis (sounds like the latter, and that's fine with me at this point). The RWMutex was there for the Trace Attributes specifically since I thought there might be more reads than writes there, but absent a contention profile I'm fine keeping it simple.